test/c++/nda_basic.cpp

```diff
@@ -22,6 +22,13 @@ static_assert(nda::is_scalar_for_v<int, matrix<std::complex<double>>> == 1, "oop
 TEST(NDA, Create1) { //NOLINT
   nda::array<long, 2> A(3, 3);
   EXPECT_EQ(A.shape(), (nda::shape_t<2>{3, 3}));
+
+  nda::array<long, 2> B({3, 3});
```

Comment:
This doesn't work on GCC 10.
```c++
/tmp/nda/test/c++/nda_basic.cpp:26:31: error: converting to ‘nda::basic_array<long int, 2, nda::C_layout, 'A', nda::heap>’ from initializer list would use explicit constructor ‘nda::basic_array<ValueType, Rank, Layout, Algebra, ContainerPolicy>::basic_array(Int ...) [with Int = {int, int}; ValueType = long int; int Rank = 2; Layout = nda::C_layout; char Algebra = 'A'; ContainerPolicy = nda::heap]’
   26 |   nda::array<long, 2> B({3, 3});
      |   
```

Reply:
I think this was supposed to be
```suggestion
  nda::array<long, 2> B(std::array<long, 2>{3, 3});
```
but the bare initialization list `{3, 3}` cannot be implicitly converted here because the constructor from `std::array` is marked `explicit`.

Reply:
Yes, it's the right thing to construct the std::array here to avoid confusion with the initializer list constructors.
This should be fixed now. It's still strange that this compiles with clang and not gcc.